### DISCUSSION TITLE
How to define "route.name" typescript with React Navigation 7 ?

### DISCUSSION THREAD

**Discussion by GaylordP** [react-navigation/react-navigation#12314] [Q&A] [answered]
Hello, and thank you for your library! :)

Do you know how to type "route.name" with React Navigation 7 ?
[I followed the documentation to type the parameters](https://reactnavigation.org/docs/typescript/), but I can’t figure out how to type the route name. For example:

```
export type RoomScreenProps = StaticScreenProps<{
  id: RoomType['id'];
}>;

const RoomScreen = ({ route }: RoomScreenProps) => {
  console.log(route.name);
};
```

> The ‘name’ property does not exist on the type ‘{ params: { id: number; }; }’.ts(2339)

Everything works perfectly otherwise, and the parameters (route.params) are correctly detected across all my screens, so I believe my project is properly set up.

Thank you in advance! :)

**Comment by satya164** [comment, marked as answer]
I'd suggest using `useRoute().name` to access the name instead. It's not supported specifying the name in props since it's inferred from the property name in static config. So the generated param list will have proper names.

**Comment by GaylordP** [comment]
Thanks very much :)
